### PR TITLE
Discard wheel dep subgraphs

### PR DESCRIPTION
**Pull request by kislyuk**

### CODE REVIEW

File: scripts/find_missing_wheels.py [hunk 1]
```diff
@@ -19,6 +19,13 @@ def build_wheel(wheel_identifier, wheels_dir):
     subprocess.check_output(["pip", "download", wheel_identifier], cwd=wd)
     if glob.glob(os.path.join(wd, "*.tar.gz")):
         subprocess.check_output("pip wheel *.tar.gz && rm -f *.tar.gz", shell=True, cwd=wd)
+    # The sub-dependency wheels downloaded and built here do not obey the pins set by pip freeze in the overall
+    # requirements.txt. We only keep the top level wheel. Since requirements.txt contains the whole flattened dependency
+    # graph, each missing sub-dependency wheel is handled at top level in the overall scan.
+    for wheel_filename in os.listdir(wd):
+        if not re.match(wheel_identifier.replace("-", "_").replace("==", "-"), wheel_filename):
```
Comment (ttung) [inline scripts/find_missing_wheels.py]: Why not just `str.find()`?  `re` could have unexpected interactions.

Comment (kislyuk) [reply scripts/find_missing_wheels.py]: Good point (although wheel names are sanitized to alphanumeric + `_`).

I'm going to use the `if substring not in string:` form if you don't mind.

Comment (kislyuk) [reply scripts/find_missing_wheels.py]: Actually, scratch that, I want to anchor at 0, so I'll use `if str.find(sub) != 0`.

**Comment by kislyuk** [comment]
Removing complex expression from f-string, in the interest of holding myself accountable :)

**Pull request merged**